src/test/java/nextstep/subway/acceptance/SectionAcceptanceTest.java

```diff
@@ -36,20 +40,124 @@ public void setUp() {
     }
 
     /**
-     * When 지하철 노선에 새로운 구간 추가를 요청 하면
-     * Then 노선에 새로운 구간이 추가된다
+     * When 새로운 역을 하행 종점으로 등록할 경우
+     * Then 새로운 두 개의 구간이 생성된다.
+     *   - 추가된 역이 해당 노선의 하행 종점역이 된다.
+     *   - 노선의 총 길이 = 기존 구간 길이 + 새로운 구간 길이
      */
-    @DisplayName("지하철 노선에 구간을 등록")
+    @DisplayName("지하철 구간 등록 - 정상1 : 새로운 역이 하행 종점인 구간 추가")
     @Test
-    void addLineSection() {
+    void addLineSection_ValidCase1() {
         // when
         Long 정자역 = 지하철역_생성_요청("정자역").jsonPath().getLong("id");
-        지하철_노선에_지하철_구간_생성_요청(신분당선, createSectionCreateParams(양재역, 정자역));
+        지하철_노선에_지하철_구간_생성_요청(신분당선, createSectionCreateParams(양재역, 정자역, sectionDistance2));
 
         // then
         ExtractableResponse<Response> response = 지하철_노선_조회_요청(신분당선);
         assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
         assertThat(response.jsonPath().getList("stations.id", Long.class)).containsExactly(강남역, 양재역, 정자역);
+        assertThat(response.jsonPath().getInt("length")).isEqualTo(sectionDistance1 + sectionDistance2);
+
+    }
+
+    /**
+     * When 새로운 역을 상행 종점으로 등록할 경우
+     * Then 새로운 두 개의 구간이 생성된다.
+     *   - 추가된 역이 해당 노선의 상행 종점역이 된다.
+     *   - 노선의 총 길이 = 기존 구간 길이 + 새로운 구간 길이
+     */
+    @DisplayName("지하철 구간 등록 - 정상2 : 새로운 역이 상행 종점인 구간 추가")
+    @Test
+    void addLineSection_ValidCase2() {
+        // when
+        Long 신논현역 = 지하철역_생성_요청("신논현역").jsonPath().getLong("id");
+        지하철_노선에_지하철_구간_생성_요청(신분당선, createSectionCreateParams(신논현역, 강남역, sectionDistance2));
+
+        // then
+        ExtractableResponse<Response> response = 지하철_노선_조회_요청(신분당선);
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
+        assertThat(response.jsonPath().getList("stations.id", Long.class)).containsExactly(신논현역, 강남역, 양재역);
+        assertThat(response.jsonPath().getInt("length")).isEqualTo(sectionDistance1 + sectionDistance2);
+    }
+
+    /**
+     * When 생성된 구간의 상행역 또는 하행역을 기준으로 그 사이에 새로운 구간을 추가하면
+     *   - 단, 새로운 구간의 길이는 기존 구간의 길이보다 작아야한다.
+     * Then 새로운 두 개의 구간이 생성된다. (상행역 - 추가된역 & 추가된역 - 하행역)
+     *   - 노선의 총 길이 = 기존 구간의 길이
```

Comment:
주석을 작성할때도 OOP 패러다임이 적용될 수 있습니다. 
설명이 로직에 대한 의존성이 늘어나면 나중에 유지보수 시점에서 관리해야할 포인트가 늘어나는 문제가 생기죠. 
그렇기에 주석은 최대한 추상적으로 how가 아닌 what을 작성하는게 좋습니다. 

```suggestion
     * When 노선 중간에 구간을 추가할 경우
     * Then 새로운 구간이 생성된다. 
```

이정도로도 충분할 것 같습니다. (의도만 포함한거라 다르게 작성할 수 있습니다. )

Reply:
주석도 유지보수 대상이라는 생각을 못했었는데 좋은 포인트 감사합니다 !